pandas/core/arrays/categorical.py

```diff
@@ -1971,10 +1971,14 @@ def __setitem__(self, key, value):
                 raise ValueError("Cannot set a Categorical with another, "
                                  "without identical categories")
 
-        rvalue = value if is_list_like(value) else [value]
-
         from pandas import Index
-        to_add = Index(rvalue).difference(self.categories)
+        if isinstance(value, tuple):
+            rvalue = [value]
```

Comment:
so the big problem I have here is that this should be a TypeError if the categories are not actual tuples. We dont' have an easy / reliable way to check this as its just an object Index. This is adding more and more technical debt on top of tupleize_cols.

Reply:
I'm gonna look into this more over the weekend / next week.

Thinking of sets, but want to understand `tupleize_cols` before I go another way.